dcicutils/trace_utils.py

```diff
@@ -0,0 +1,56 @@
+import functools
+
+from dcicutils.misc_utils import environ_bool, full_object_name, get_error_message, PRINT
+from dcicutils.obfuscation_utils import obfuscate_dict
+
+
+# You can use TRACE_REDACT=FALSE if tracing should show credentials,
+# but obviously this is usually not something you'd want to do, so this
+# is on a separate variable that defaults to True. -kmp 30-Mar-2023
+
+TRACE_REDACT = environ_bool("TRACE_REDACT", default=True)
+
+
+def _maybe_obfuscate(x):
```

Comment:
Note in the case of SubmitCGAP this will print the value of the access secret in plaintext, as part of the `auth` tuple. Not sure we want a special case for this.